core/util.py

```diff
@@ -0,0 +1,34 @@
+from time import sleep
+
+
+def retry(func) -> bool:
+    attempts = 5
+    attempt = 0
+    while True:
+        if func():
+            break
+        else:
+            attempt += 1
+            if attempt >= attempts:
+                raise Exception(f"Operation failed after {attempts}.")
+            sleep(1)
+            continue
+
+    return True
+
+
+def recursively_replace_value(mydict: dict, key: str, val: str):
+    """
+    Recursively searches mydict for key and replaces its value (if any) with val
+    :param mydict:
+    :param key:
+    :param val:
+    :return:
+    """
+    if key in mydict:
```

Comment:
Imaginary edge case: What if `key` is not a `str` but a list or dict, should we safeguard against replacing such values?

Reply:
It's case I don't think we need to worry about.